backend/src/api/get_files.rs

```diff
@@ -0,0 +1,74 @@
+use crate::{PublicFileMetadata, State};
+use ic_cdk::export::candid::Principal;
+
+pub fn get_files(state: &State, caller: Principal) -> Vec<PublicFileMetadata> {
+    match state.file_owners.get(&caller) {
+        None => vec![],
+        Some(file_ids) => file_ids
+            .iter()
+            .map(|file_id| PublicFileMetadata {
+                file_id: *file_id,
+                file_name: state
+                    .file_data
+                    .get(file_id)
+                    .unwrap()
```

Comment:
I'd add a message to communicate why this isn't expect to fail

```suggestion
                    .expect("file must exist")
```